moabb/evaluations/base.py

```diff
@@ -1,33 +1,121 @@
+import logging
 from abc import ABC, abstractmethod
-import numpy as np
+
+from sklearn.base import BaseEstimator
+
+from moabb.analysis import Results
+from moabb.datasets.base import BaseDataset
+from moabb.paradigms.base import BaseParadigm
+
+log = logging.getLogger()
 
 
 class BaseEvaluation(ABC):
     '''Base class that defines necessary operations for an evaluation.
     Evaluations determine what the train and test sets are and can implement
     additional data preprocessing steps for more complicated algorithms.
 
-    random_state: if not None, can guarantee same seed
-    n_jobs: 1; number of jobs for fitting of pipeline
+    Parameters
+    ----------
+    paradigm : Paradigm instance
+        the paradigm to use.
+    datasets : List of Dataset Instance.
+        The list of dataset to run the evaluation. If none, the list of
+        compatible dataset will be retrieved from the paradigm instance.
+    random_state:
+        if not None, can guarantee same seed
+    n_jobs: 1;
+        number of jobs for fitting of pipeline
 
     '''
 
-    def __init__(self, random_state=None, n_jobs=1):
+    def __init__(self, paradigm, datasets=None, random_state=None, n_jobs=1):
         """
         Init.
         """
-        if random_state is None:
-            self.random_state = np.random.randint(0, 1000, 1)[0]
+
+        self.random_state = random_state
```

Comment:
Allowing random_state to be none means that splits in cross-validation are not identical for all pipelines (in WithinSessionEvaluation as of now, others have defined splits)

Reply:
Duh .... i dont know what i was thinking. I will revert this